src/Google.Cloud.Functions.Invoker/FunctionsExtensions.cs

```diff
@@ -0,0 +1,175 @@
+        /// implement one of the Functions Framework function interfaces. 
+        /// </summary>
+        /// <remarks>
+        /// If this method completes successfully, a binding for <see cref="IHttpFunction"/> will definitely
+        /// have been added to the service collection. Other bindings may also be present, in order to adapt
+        /// the function to <see cref="IHttpFunction"/>.
+        /// </remarks>
+        /// <param name="services">The service collection to configure.</param>
+        /// <param name="type">The target function type.</param>
+        /// <returns>The original service collection, for method chaining.</returns>
+        public static IServiceCollection AddFunctionTarget(this IServiceCollection services, Type type) =>
+            HostingInternals.AddServicesForFunctionTarget(services, type);
+
+        /// <summary>
+        /// Configures the web host builder to use the functions startup classes specified by
+        /// assembly attributes.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="assembly">The assembly to query for attributes specifying startup classes.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionsStartups(this IWebHostBuilder builder, Assembly assembly)
+        {
+            foreach (var startup in HostingInternals.GetStartups(assembly))
+            {
+                builder.AddFunctionsStartup(startup);
+            }
+            return builder;
+        }
+
+        /// <summary>
+        /// Configures the web host builder to use the specified functions startup classes. This method can be called
+        /// multiple times, and the startups will all be configured.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <param name="startup">The startup class to use.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder AddFunctionsStartup(this IWebHostBuilder builder, FunctionsStartup startup) =>
+            builder.ConfigureServices((context, services) => HostingInternals.AddStartup(context, services, startup));
+
+        /// <summary>
+        /// Configures the given web host builder to use the Functions Framework.
+        /// This method executes any application configurers before adding handlers for fixed paths to
+        /// return "not found" responses for fixed paths (e.g. "favicon.ico") and setting the terminal
+        /// handler to execute the target function.
+        /// </summary>
+        /// <param name="builder">The web host builder to configure.</param>
+        /// <returns>The original builder, for method chaining.</returns>
+        public static IWebHostBuilder UseFunctionsFramework(this IWebHostBuilder builder) =>
```

Comment:
I think I would add some remarks saying all the other things that **need** to be configured before calling this one, I think at least the configuration source and the target function. People might get the wrong idea that this does everything with some defaults.

Reply:
Good plan. Alternatively, we could make *this* method do "the lot" and put a separate extension method on IApplicationBuilder.

I believe that's what you suggested in #118, and I dismissed it at the time - but I've thought of a good use case now, which is "I want to add a new configuration provider, but otherwise use normal startup".

It's possible that having EntryPoint methods accepting configuration delegates for that would cover that. Let's talk about it more before merging - I think I'm going to write docs with sample use cases before merging too, to make sure everything is feasible.